openml-h2o/src/main/java/com/feedzai/openml/h2o/H2OModelCreator.java

```diff
@@ -160,7 +154,9 @@ public DatasetSchema loadSchema(final Path modelPath) throws ModelLoadingExcepti
         errorBuilder.addAll(ValidationUtils.validateModelInDir(modelPath));
         errorBuilder.addAll(ValidationUtils.checkNoFieldsOfType(schema, StringValueSchema.class));
 
-        ValidationUtils.validateCategoricalSchema(schema).ifPresent(errorBuilder::add);
+        if (schema.getTargetIndex().isPresent()) {
+            ValidationUtils.validateCategoricalSchema(schema).ifPresent(errorBuilder::add);
+        }
```

Comment:
This way we are validating the schema depending on itself. Instead, we should be validating the schema conditioned to the model type (e.g., if supervised, force this validation, if unsupervised, do not).

Reply:
Note that at this point we do not have a model to validate against. I agree with you, but note the cyclicity here:
1) We want to load a model
2) we need to validate the "model loading" arguments
3) one of the arguments is the schema. In order to validate the schema we then need the model that we are trying to load in 1)

Reply:
Ok then what about: where do we ensure that an Unsupervised Model is not loaded with a DatasetSchema that has a target variable? Shouldn't be possible.

Reply:
You have two points here:
 - I think that it shouldn't be possible to have a schema with no target assigned to a supervised model. Will add the necessary validations for that, but basically we have to validate it after loading the `GenModel` and before creating the "OpenML" model.
 - Regarding the opposite (having a schema with target assigned to an unsupervised), it really depends on the algorithm from my point of view. The implementation (and tests) at this stage allow (and test) training an Isolation Forest model with target field, as H2O actually allows it.

Reply:
Ok, so long as that is documented on the javadoc of each type of model :)